main-settings/src/main/scala/sbt/unchecked.scala

```diff
@@ -0,0 +1,13 @@
+package sbt
+
+import scala.annotation.Annotation
+
+/** An annotation to designate that the annotated entity
+ *  should not be considered for additional sbt compiler checks.
+ *  These checks ensure that the DSL is predictable and prevents
+ *  users from doing dangerous things at the cost of a stricter
+ *  code structure.
+ *
+ *  @since 1.0.0
+ */
+class unchecked extends Annotation
```

Comment:
The reason I preferred to create our own unchecked annotation is because reusing the Scala one can cause more confusion, and it's used for different purposes so I don't want to overload its semantics. The good thing is that `scala.annotation.unchecked` is by default not in the scope of sbt files, but `sbt.unchecked` is -- so you can have a seamless experience. Plugin files will need to import explicitly.
  
I think that keeping the same name as the Scala one is good for people to remember it.  They share the name, but not the namespace. Otherwise, it's a tedious task to remember "that special annotation that disables the sbt checks".